### ISSUE TITLE
Muzei usually stuck for a while when I skip a wallpaper.

### ISSUE THREAD

**Issue by maple3142** [romannurik/muzei#538]
Muzei usually stuck for a while when I skip a wallpaper.
It also killed almost all background apps process.

Logcat:
```
Performing pause of activity that is not resumed: {net.nurik.roman.muzei/com.google.android.apps.muzei.MuzeiActivity}
java.lang.RuntimeException: Performing pause of activity that is not resumed: {net.nurik.roman.muzei/com.google.android.apps.muzei.MuzeiActivity}
	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3865)
	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3853)
	at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3827)
	at android.app.ActivityThread.-wrap15(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1602)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6499)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```
```
Performing stop of activity that is already stopped: {net.nurik.roman.muzei/com.google.android.apps.muzei.MuzeiActivity}
java.lang.RuntimeException: Performing stop of activity that is already stopped: {net.nurik.roman.muzei/com.google.android.apps.muzei.MuzeiActivity}
	at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:3997)
	at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4085)
	at android.app.ActivityThread.-wrap24(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1623)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6499)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by ianhanniballake** [comment]
I'll need a little more information as those stack traces don't say anything about what is actually going on.
- What version of Muzei are you using? 
- What source do you have selected?
- What version of Android are you using?

**Comment by maple3142** [comment]
Muzei: 2.5.2
Source: https://play.google.com/store/apps/details?id=moe.democyann.pixivformuzeiplus
Android: 8.1.0

**Comment by ianhanniballake** [comment]
The current Muzei API requires that you disable Battery Optimizations on third party sources (such as Pixiv for Muzei Plus) for them to continue to have network connectivity to load more wallpapers. Work is being done on a new API that removes this requirement.

**Comment by Xalaxis** [comment]
@ianhanniballake In the interim, could a message perhaps be added to the app to inform users of this?  I'm aware a note has been added to the description but believe it can easily be missed (see Play store reviews as a good example).  Perhaps the note could also be moved further up the description?

**Comment by ianhanniballake** [comment]
A warning was added in f5f62b5 and will be available in a future Muzei 3.0 alpha
